packages/rate-limit-controller/src/RateLimitController.ts

```diff
@@ -184,16 +204,19 @@ export class RateLimitController<
     const rateLimitTimeout =
       this.implementations[api].rateLimitTimeout ?? this.rateLimitTimeout;
     this.update((state) => {
-      // TODO: Replace `any` with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const previous = (state as any).requests[api][origin] ?? 0;
-      // TODO: Replace `any` with type
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (state as any).requests[api][origin] = previous + 1;
-
+      const previous =
+        (state as unknown as RateLimitState<RateLimitedApis>).requests[api][
+          origin
+        ] ?? 0;
       if (previous === 0) {
         setTimeout(() => this.resetRequestCount(api, origin), rateLimitTimeout);
       }
+      return Object.assign({}, state, {
+        requests: {
+          ...(state.requests as RateLimitedRequests<RateLimitedApis>),
+          [api]: { [origin]: previous + 1 },
+        },
+      });
     });
```

Comment:
Do we need to be inside of the `update` to access the `state` to set `previous`? It seems like we can take away one of the type assertions here:

``` suggestion
    const previous = this.state.requests[api][origin] ?? 0;
    this.update((state) => {
      if (previous === 0) {
        setTimeout(() => this.resetRequestCount(api, origin), rateLimitTimeout);
      }
      return Object.assign({}, state, {
        requests: {
          ...(state.requests as RateLimitedRequests<RateLimitedApis>),
          [api]: { [origin]: previous + 1 },
        },
      });
    });
```

Reply:
Also, is the type on `origin` correct? I tried to change the `update` call so that we are directly modifying state (well, the draft state) instead of returning an object, but the type on `origin` seemed to be interfering for some reason.

Reply:
Setting `previous` outside of `update` applied here: https://github.com/MetaMask/core/pull/3963/commits/71cc83dfab8d6c3ef7ff99bdfbbdadef9a07e9e4

Mutating state in-place seems to work with `Object.assign`: https://github.com/MetaMask/core/pull/3963/commits/49e6e05da30ffc5d50b00dbdf68fee334b3b35b1